dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -50,6 +54,18 @@ private static Map<String, ArgumentMetadata> getArgumentMap(AnnotatedMethod<?> m
                 }
             }
         }
-        return result;
+        return result.isEmpty() ? Collections.emptyMap() : Map.copyOf(result);
+    }
+
+    private static Map<String, SpecialArgumentMetadata> getSpecialArgumentMap(AnnotatedMethod<?> method) {
+        Map<String, SpecialArgumentMetadata> result = new HashMap<>();
+        for (AnnotatedParameter<?> p : method.getParameters()) {
+            ToolArg pInfo = p.getAnnotation(ToolArg.class);
+            if (pInfo == null) {
+                SpecialArgumentMetadata pData = new SpecialArgumentMetadata(SpecialArgumentType.fromClass(p.getBaseType()), p.getPosition());
+                result.put(p.getBaseType().getTypeName(), pData);
+            }
+        }
```

Comment:
Here we're assuming that anything not a tool argument is a special argument. We should validate the special arguments. I'll raise an issue for that.